ostree-upload-server.py

```diff
@@ -93,54 +64,101 @@ def get_count(self):
 
 
 class UploadWebApp(Flask):
-    def __init__(self, import_name, upload_counter, webapp_callback):
+    def __init__(self, import_name, users, repo, upload_counter, push_adapters, webapp_callback):
         super(UploadWebApp, self).__init__(import_name)
+        self._users = users
+        self._repo = repo
         self._upload_counter = upload_counter
+        self._push_adapters = push_adapters
         self._webapp_callback = webapp_callback
 
         self.route("/")(self.index)
         self.route("/upload", methods=["GET", "POST"])(self.upload)
+        self.route("/push")(self.push)
 
         self._tempdir = tempfile.mkdtemp(prefix="ostree-upload-server-")
         atexit.register(os.rmdir, self._tempdir)
 
+    def _check_auth(self):
+        if not self._users:
+            return True
+        auth = request.authorization
+        if auth and \
+                auth.username in self._users and \
+                auth.password == self._users.get(auth.username):
```

Comment:
Oh, I see. No, that's good enough. I can always not expose the /push endpoint, too.